File: week14/19/P1.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

class Solution {
public:
    string intToRoman(int num) {
        string answer = "";
        vector<string> roman_s = {"M","CM","D","CD","C","XC","L","XL","X","IX","V","IV","I"};
        vector<int> roman_i = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};
        while (num > 0) {
            for (int r=0; r<13; r++) {
                if (num >= roman_i[r]) {
                    answer.append(roman_s[r]);
                    num -= roman_i[r];
                    break;
                }
            }
        }
        return answer;
    }
};

int main() {
    Solution result;
    int num1 = 3749;
    cout << result.intToRoman(num1) << endl;
    int num2 = 58;
    cout << result.intToRoman(num2) << endl;
    int num3 = 1994;
    cout << result.intToRoman(num3) << endl;

    return 0;
}

File: week14/19/P1.py
class Solution(object):
    def intToRoman(self, num):
        answer = ""
        roman = {1000:"M", # python 3.5 이하에선 순서가 보장되지 않음
                 900:"CM", 500:"D", 400:"CD", 100:"C",
                 90:"XC", 50:"L", 40:"XL", 10:"X",
                 9:"IX" , 5:"V", 4:"IV", 1:"I"}
        while (num > 0):
            for r in roman:
                if (num >= r):
                    answer += roman[r]
                    num -= r
                    break
        return answer

if __name__ == "__main__":
    result = Solution()
    num = 3749
    print(result.intToRoman(num))
    num = 58
    print(result.intToRoman(num))
    num = 1994
    print(result.intToRoman(num))
